dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/AbstractImageRegistry.java

```diff
@@ -240,14 +249,114 @@ public Tool refreshTool(final long toolId, final Long userId, final UserDAO user
+
+        Map<String, List<Map<String, List<Map<String, String>>>>> manifestDigestMap = new HashMap<>();
+        Map<String, List<Map<String, String>>> nameMap = new HashMap<>();
+        Map<String, String> error = new HashMap<>();
+
+        if (dockerHubResponse.isPresent()) {
+            Gson gson = new Gson();
+            final String errorJSON = dockerHubResponse.get();
+            final String manifestJSON = dockerHubResponse.get();
+            final String nameJSON = dockerHubResponse.get();
+
+            error = (Map<String, String>)gson.fromJson(errorJSON, error.getClass());
+            if (error.get("message") != null) {
+                LOG.info("Error response from DockerHub: " + error.get("message"));
+                return new ArrayList<>();
+            }
+
+            manifestDigestMap = (Map<String, List<Map<String, List<Map<String, String>>>>>)gson.fromJson(manifestJSON, manifestDigestMap.getClass());
+            nameMap = (Map<String, List<Map<String, String>>>)gson.fromJson(nameJSON, nameMap.getClass());
+
+            List<Map<String, String>> listOfTags = nameMap.get("results");
+
+            Iterator<Map<String, String>> iterator = listOfTags.iterator();
+            Iterator<Map<String, List<Map<String, String>>>> iterator2 = manifestDigestMap.get("results").iterator();
+
+            while (iterator.hasNext() && iterator2.hasNext()) {
+                Map<String, String> i = iterator.next();
+                final String tagName = i.get("name");
+                final String imageID = i.get("image_id");
+                try {
+                    final String manifestDigest = iterator2.next().get("images").get(0).get("digest");
+                    final Tag tag = new Tag();
+                    tag.setName(tagName);
+
+
+                    List<Checksum> checksums = new ArrayList<>();
+                    checksums.add(new Checksum(manifestDigest.split(":")[0], manifestDigest.split(":")[1]));
+                    Set<Image> images = Collections.singleton(new Image(checksums, repo, tag.getName(), imageID));
+                    tag.getImages().addAll(images);
+
+                    tags.add(tag);
+                } catch (IndexOutOfBoundsException ex) {
+                    LOG.info("Unable to grab image and checksum information for" + tool.getNamespace() + '/' + tool.getName());
+                }
+
+            }
+            return tags;
+        } else {
+            LOG.info("Could not get response from DockerHub");
+            return new ArrayList<>();
+        }
+    }
+
+
+    private Optional<String> getDockerHubToolAsString(Tool tool) {
+        final String repo = tool.getNamespace() + '/' + tool.getName();
+        final String repoUrl = DOCKERHUB_URL + "repositories/" + repo + "/tags";
+
+        String command = "curl " + repoUrl;
+        Optional<String> response;
+        try {
+            Process process = Runtime.getRuntime().exec(command.split(" "));
```

Comment:
Avoid `exec`, it has a number of pitfalls. 
We've wrapped commons exec https://commons.apache.org/proper/commons-exec/tutorial.html which is a bit safer in https://github.com/dockstore/dockstore/blob/423cf2269d70b4e714fee51c8b722d5692e4385a/dockstore-common/src/main/java/io/dockstore/common/Utilities.java#L124

But in this specific case, you should be able to avoid curl with something like https://guava.dev/releases/20.0/api/docs/com/google/common/io/Resources.html#copy-java.net.URL-java.io.OutputStream-